taskt/UI/Forms/frmScriptBuilder.cs

```diff
@@ -496,463 +510,329 @@ private void lstScriptActions_KeyDown(object sender, KeyEventArgs e)
                 lstScriptActions_DoubleClick(null, null);
             }
             else if ((e.Control) && (e.KeyCode == Keys.X))
-            {     
+            {
                 CutRows();
             }
             else if ((e.Control) && (e.KeyCode == Keys.C))
             {
                 CopyRows();
             }
             else if ((e.Control) && (e.KeyCode == Keys.V))
-            {        
+            {
                 PasteRows();
             }
             else if ((e.Control) && (e.KeyCode == Keys.Z))
             {
-
                 UndoChange();
-
             }
             else if ((e.Control) && (e.KeyCode == Keys.R))
             {
-
                 RedoChange();
-
             }
-
             else if ((e.Control) && (e.KeyCode == Keys.A))
             {
-
                 foreach (ListViewItem item in lstScriptActions.Items)
                 {
                     item.Selected = true;
                 }
+            }
+        }
+
+        private void lstScriptActions_DoubleClick(object sender, EventArgs e)
+        {
+            if (lstScriptActions.SelectedItems.Count != 1)
+            {
+                return;
+            }
+
+            //bring up edit mode to edit the action
+            ListViewItem selectedCommandItem = lstScriptActions.SelectedItems[0];
+
+            //set selected command from the listview item tag object which was assigned to the command
+            var currentCommand = (ScriptCommand)selectedCommandItem.Tag;
+
+            //check if editing a sequence
+            if (currentCommand is SequenceCommand)
+            {
+                if (_editMode)
+                {
+                    MessageBox.Show("Embedding Sequence Commands within Sequence Commands not yet supported.");
+                    return;
+                }
+
+                //get sequence events
+                SequenceCommand sequence = (SequenceCommand)currentCommand;
+                frmScriptBuilder newBuilder = new frmScriptBuilder();
+
+                //add variables
+                newBuilder._scriptVariables = new List<ScriptVariable>();
```

Comment:
How are we setting _scriptVariables here in the newBuilder? It's private so I would expect this to throw an exception when it's accessed.

Reply:
So this is odd. I ran/stepped through it and it didn't get an exception. Maybe because it's accessing within a method inside of the frmScriptBuilder class. Additionally, what this is being used for, SequenceCommand, is an interesting command that groups several commands into one. Kind of like RunTaskCommand but without the arguments. Instead it passes the entire variable list.

Reply:
Also those variables were set to private to begin with, I just renamed them.